packages/utils/src/ast.ts

```diff
@@ -0,0 +1,300 @@
+          if (
+            body.type === 'VariableDeclaration' &&
+            (body.declarations[0].id as Identifier).name ===
+              'regeneratorRuntime'
+          ) {
+            return;
+          }
+
+          program?.node.body.unshift(regeneratorRuntimeWrapper());
+        }
+      }
+    },
+
+    TemplateLiteral(path) {
+      const { node } = path;
+
+      // This checks if the template literal was visited before. Without this,
+      // it would cause an infinite loop resulting in a stack overflow. We can't
+      // skip the path here, because we need to visit the children of the node.
+      if (path.getData('visited')) {
+        return;
+      }
+
+      // Break up tokens that could be parsed as HTML comment terminators, or
+      // `import()` statements.
+      // For reference:
+      // - https://github.com/endojs/endo/blob/70cc86eb400655e922413b99c38818d7b2e79da0/packages/ses/error-codes/SES_HTML_COMMENT_REJECTED.md
+      // - https://github.com/MetaMask/snaps-skunkworks/issues/505
+      const expressions = node.quasis.reduce<Expression[]>(
+        (acc, quasi, index) => {
+          // Note: Template literals have two variants, "cooked" and "raw". Here
+          // we just use the raw version, but this might break code that uses
+          // the cooked version. For reference:
+          // https://exploringjs.com/impatient-js/ch_template-literals.html#template-strings-cooked-vs-raw
+          const replacement = breakTokens(quasi.value.raw).map((token) =>
+            stringLiteral(token),
+          );
+
+          if (node.expressions[index]) {
+            return [
+              ...acc,
+              ...replacement,
+              node.expressions[index],
+            ] as Expression[];
+          }
+
+          return [...acc, ...replacement];
+        },
+        [],
+      );
+
+      path.replaceWith(
+        templateLiteral(
+          new Array(expressions.length + 1)
+            .fill(undefined)
+            .map(() => templateElement({ cooked: '', raw: '' })),
+          expressions,
+        ) as Node,
+      );
+
+      path.setData('visited', true);
+    },
+
+    StringLiteral(path) {
+      const { node } = path;
+
+      // Break up tokens that could be parsed as HTML comment terminators, or
+      // `import()` statements.
+      // For reference:
+      // - https://github.com/endojs/endo/blob/70cc86eb400655e922413b99c38818d7b2e79da0/packages/ses/error-codes/SES_HTML_COMMENT_REJECTED.md
+      // - https://github.com/MetaMask/snaps-skunkworks/issues/505
+      const tokens = breakTokens(node.value);
+      const replacement = tokens.slice(1).reduce<Expression>(
+        (acc, value) => binaryExpression('+', acc, stringLiteral(value)),
+        // In the case of an empty string, `tokens[0]` will be undefined, so
+        // we provide a fallback value of an empty string.
+        stringLiteral(tokens[0] ?? ''),
```

Comment:
Maybe this is what changes `'` to `"`? 🤔 

In general we could just ignore strings that don't need replacement?